src/phoenix/datasets/experiments.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import datetime
+from typing import (
+    Any,
+    Callable,
+    Coroutine,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    TypedDict,
+    Union,
+)
+
+import httpx
+
+from phoenix.config import (
+    get_env_collector_endpoint,
+    get_env_host,
+    get_env_port,
+)
+from phoenix.evals.executors import get_executor_on_sync_context
+from phoenix.evals.models.rate_limiters import RateLimiter
+
+JSONSerializable = Union[Dict[str, Any], List[Any], str, int, float, bool, None]
+ExperimentCallable = Callable[[JSONSerializable], JSONSerializable]
+ExperimentCoroutineFn = Callable[[JSONSerializable], Coroutine[Any, Any, JSONSerializable]]
+
+
+class ExampleProtocol(Protocol):
+    @property
+    def id(self) -> str: ...
+
+    @property
+    def input(self) -> JSONSerializable: ...
```

Comment:
Examples also have output and metadata.

Reply:
protocols can be incomplete, these are stubs for the actual clientside model objects we're going to build

Reply:
I guess we've only got the input here since that is all that's used for creating a run.